### PR TITLE
Add srf module

### PR DESCRIPTION
**Pull request by lispandfound**
# What is this module?
This module allows for both reading and writing SRF files.
# Why not use the module in QCore?
For a few reasons:
1. The `qcore.srf` module does not support writing SRF files.
2. Exposing SRF points as a pandas dataframe allows manipulation of the points using efficient vectorised operations. We use this in rupture propagation to delay ruptures by adding to the `tinit` column.
3. There is better documentation for the new module than the old one.
# Example

```python
>>> srf_file = srf.read_srf('/path/to/srf')
>>> srf_file.points['tinit'].max() # get the last time any point in the SRF ruptures
>>> srf_file.points['tinit'] += 1 # delay all points by one second
>>> coordinates.wgs_depth_to_nztm(srf_file.header[['elat', 'elon']].to_numpy())
#   ^ get the coordinates all the fault plane centres in the rupture in NZTM format
# etc...
>>> srf.write_srf('/path/to/srf', srf_file)
```

### CODE REVIEW

File: source_modelling/srf.py [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+    srf_ffp : Path
+        The filepath of the SRF file.
+
+    Returns
+    -------
+    SrfFile
+        The filepath of the SRF file.
+    """
+    with open(srf_ffp, mode="r", encoding="utf-8") as srf_file_handle:
+        version = read_version(srf_file_handle)
```
Comment (claudio525) [inline source_modelling/srf.py:303]: Any particular reason for making functions for `read_version` and `read_points_count` ? They are both very short, very specific (one specific task that is generally not done by itself and requires file to be at the correct location) and only used once. Is there any scenarios where a user wouldn't use `read_srf`?

File: source_modelling/srf.py [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+
+    Parameters
+    ----------
+    srf_ffp : Path
+        The filepath to write the srf object to.
+    srf : SrfFile
+        The SRF object.
+    """
+    with open(srf_ffp, mode="w", encoding="utf-8") as srf_file_handle:
+        write_version(srf_file_handle)
```
Comment (claudio525) [inline source_modelling/srf.py:404]: Same comment as above.

Comment (claudio525) [reply source_modelling/srf.py:404]: In my opinion this module should only have 2 or maybe 4 functions, unless there is a good reason for splitting it up this much?

Comment (lispandfound) [reply source_modelling/srf.py:404]: It's a legacy of a slightly older design. I can easily change that.

Comment (claudio525) [reply source_modelling/srf.py:404]: Cool, I think that would be a good idea.

**Review by claudio525 — APPROVED** [review]

**Review by joelridden — APPROVED** [review]

**Review by sungeunbae — APPROVED** [review]

**Review by AndrewRidden-Harper — APPROVED** [review]
nice description of the differences with qcore.srf

**Pull request merged**